test/unit/range/iterator_test_template.hpp

```diff
@@ -54,14 +55,20 @@ SEQAN3_CONCEPT has_expect_equal_member_function = requires(t & a)
 
 // Delegates to the test fixture member function `expect_eq` if available and falls back to EXPECT_EQ otherwise.
 template <typename T, typename A, typename B>
-void expext_eq(A && a, B && b)
+void expect_iter_value_equal(A && a, B && b)
 {
     if constexpr (has_expect_equal_member_function<iterator_fixture<T>>)
         iterator_fixture<std::remove_reference_t<T>>::expect_eq(a, b);
     else
         EXPECT_EQ(a, b);
 }
 
+template <typename T, typename it_t, typename rng_it_t>
+void expect_iter_equal(it_t && it, rng_it_t && rng_it)
+{
+    expect_iter_value_equal<T>(*it, *rng_it);
```

Comment:
This is not possible in a generic way otherwise we wouldn't need this function at all.

`rng_it` might be a `std::vector::iterator` and `it_t` might be some `input_iterator`. Two types that are not comparable with each other out of the box.

The whole idea of that function `expect_iter_equal` is to allow a (customisable) comparison between two unrelated iterator types. We assume in our tests that two (unrelated) iterators are equal, iff they have the same value. In a later PR, I will add the ability to customise this function, so that I can ask invariants about iterators. My use case: Has my iterator, which was copied or modified, the correct state / initialised / copied member value after each operation.

I personally think that the `expect_iter_equal` name is pretty spot-on, but I'm always open for better names. Here is what this function does: 

Get two unrelated iterators. Both iterators are expected to be in sync with each other, because they always get the same operations applied. The first iterator is always the test subject and the second iterator always the "expected" state. This function compares the expected state (contained in the second iterator) that the first iterator should be in with the actual state of the first iterator. The default implementation assumes that the range values are the same.

The name before this change was simply `expect_equal`, so you didn't even know what was equal.

Reply:
Okay, I'm convinced 👍